cpp/inc/bond/core/blob.h

```diff
@@ -54,6 +57,8 @@ class blob
           _content(_buffer.get() + offset),
           _length(length)
     {
+        bond::detail::check_add_overflow(_buffer.get(), offset);
```

Comment:
On line 57 we've already performed the potentially overflowing addition. I think we want a function that takes a pointer and an offset, throws if it would be outof range, and returns a value otherwise. Then, we can do something like this:

```
blob(const boost::shared_ptr<const char[]>& buffer, uint32_t offset, uint32_t length)
    : _buffer(buffer),
      _content(detail::safe_add(buffer.get(), offset)),
     _length(length)
    {
        check_add_overflow(_content, length);
    }
```

Reply:
Good point.